src/importers/gcp/importer.py

```diff
@@ -0,0 +1,176 @@
+        object_name: Name of the object (file) in the bucket.
+        output_path: Local path to save the downloaded file.
+
+    Raises:
+        Exception: If any error occurs during the download process.
+    """
+    try:
+        bucket = storage_client.bucket(bucket_name)
+        blob = bucket.blob(object_name)
+        blob.download_to_filename(output_path)
+        logger.info(f"Downloaded gs://{bucket_name}/{object_name} to {output_path}")
+    except Exception as e:
+        logger.exception(f"Error downloading gs://{bucket_name}/{object_name}: {e}")
+        raise
+
+
+def process_gcs_message(
+    message: pubsub_v1.subscriber.message.Message,
+    db: object,
+) -> None:
+    """Processes an incoming Pub/Sub message from GCS.
+
+    This function handles the processing of a Google Cloud Storage (GCS)
+    object creation event received via Pub/Sub. It extracts file metadata,
+    downloads the file, creates a file record in the database, and
+    generates file hashes.
+
+    Args:
+        message: The Pub/Sub message containing GCS event data.
+        db: The database session.
+
+    Raises:
+        Exception: If any error occurs during the processing.
+    """
+    try:
+        # Acknowledge the message immediately to prevent reprocessing
+        message.ack()
+
+        # Extract GCS data from the Pub/Sub message
+        data = json.loads(message.data.decode("utf-8"))
+        bucket_name = data["bucket"]
+        object_name = data["name"]
+
+        logger.info(f"Processing GCS file: gs://{bucket_name}/{object_name}")
+
+        # Skip processing if the file is in the root bucket or is a directory
+        if "/" not in object_name:
+            logger.info("File is in root bucket, skipping import.")
+            return
+
+        if object_name.endswith("/"):
+            logger.info("GCS directory created, nothing to import.")
+            return
+
+        # Extract file metadata
+        folder_id, filename, file_extension, output_filename = extract_file_info(
+            object_name
+        )
+
+        # Get folder information from the database
+        folder = get_folder_from_db(db, folder_id)
+        if not folder:
+            logger.error(f"Folder with ID {folder_id} not found in OpenRelik.")
+            return
+
+        # Construct the output path
+        output_path = os.path.join(folder.path, output_filename)
+
+        # Download the file from GCS
+        storage_client = storage.Client(project=PROJECT_ID)
+        download_file_from_gcs(storage_client, bucket_name, object_name, output_path)
+
+        # Create file record in the database
+        new_file_db = create_file_record(
+            db,
+            filename,
+            uuid.UUID(output_filename.split(".")[0]),
+            file_extension,
+            folder.id,
+            ROBOT_ACCOUNT_USER_ID,
+        )
+
+        # Generate file hashes (consider moving this to a background job)
+        generate_hashes(new_file_db.id)
```

Comment:
Agree on this being a background job. Should this really be enabled by default on the server-side? This could get expensive and looks like a potential bottleneck for the server component which has more important functions than hashing.